### PR TITLE
[TRTLLM-14831][chore] Consolidate package bootstrap and relocate restricted deserialization

### PR DESCRIPTION
**Pull request by YihuiLu512**
Reduce the package root's mechanics to a single private bootstrap module, and move the restricted unpickler into the package that consumes it. Each change is described below together with the reasoning it rests on and what it changes for callers.

1) tensorrt_llm/_bootstrap.py (new), tensorrt_llm/__init__.py

_common.py's _init() and the three setup functions that lived in __init__.py -- DLL directory, Python-library preload, vendored-triton_kernels precedence -- merge into one private bootstrap module. __init__.py drops from 176 to 103 lines and is now only the package's compatibility surface: the intentional re-exports and __all__.

The two existing phases keep their order, which is the constraint the file is arranged around: _prepare_environment() runs before `import torch`, _init() runs after the package's own imports, and its early-return guard is unchanged.

That ordering forces the one change here that is not a pure move. _bootstrap.py is imported ahead of torch, so its module scope is restricted to the standard library; a module-level `import torch` or `from .bindings import MpiComm` would load Torch and the compiled extension before phase 1 had prepared for them. Phase 2's four imports -- torch, tensorrt_llm.bindings, ._utils and .logger -- therefore move into _init()'s body. This defers statements, not any module's first import: _init() is the last statement of __init__.py, and all four are already imported earlier on the `import tensorrt_llm._torch.models` path. Tracing module completion order with -X importtime over `import tensorrt_llm` confirms it: 2393 modules on both sides, and the whole difference is +tensorrt_llm._bootstrap at the front and -tensorrt_llm._common where it used to sit.

_common.py is absent from legacy-files.txt while __init__.py is on it, so the three relocated functions land in ruff's scope instead of yapf's and are reformatted accordingly. That is formatting only and confined to those three functions; _init() was already ruff-clean and moves unchanged.

2) tensorrt_llm/_common.py becomes a shim

The old path keeps a definition-free shim that re-exports _init and warns on import. The module is private, and every source of evidence that can be checked inside this repository is clean -- but one cannot be checked here, namely consumers outside the repository, and only the module's owner can answer for those. That question was never asked, so removing the path was never certifiable. Ten lines against a breakage nobody here can observe is the same asymmetry that keeps a shim on every public path in this migration, so the shim goes in rather than the question going to an owner.

The export set is recorded rather than computed: both names in the old module are underscored, so a private module has no computable public surface, and the shim re-exports the one symbol this repository itself consumed. _inited is deliberately left out -- it is mutable module state, and a re-export would bind the value once and then diverge from the guard it appears to mirror.

What the shim preserves is one function object, not a copy: `_common._init is _bootstrap._init is tensorrt_llm._init`. A copy would give the guard two _inited flags, and a call through the old path would run phase 2 -- custom op registration and MPI init -- a second time, which it is not designed for; calling _common._init() again is verified to be a no-op. `import tensorrt_llm` does not pull the shim in, since __init__.py now imports _bootstrap, so the DeprecationWarning still means "you named the old path" rather than firing for every user.

3) tensorrt_llm/serialization.py moves to tensorrt_llm/llmapi/serialization.py

The restricted unpickler moves into the package that consumes it, and both consumers at HEAD are rewritten to the canonical path: llmapi/rlhf_utils.py, the only production consumer, and tests/unittest/llmapi/test_serialization.py. A stale reference in a model_engine.py comment is retargeted as well. Nothing in the tree goes through the shim -- in-repo references, lint lists, ownership rules and derived configs are all rewritten as if the old path had disappeared. The file gains the NVIDIA copyright header it never had, and the four allowlist module strings travel with it unchanged; later tasks of this migration update their contents at the new location.

Ownership moves with the code. That is the one effective-CODEOWNERS change in this commit and it is intended: serialization.py inherited the package-root default and now falls under llmapi/'s rule, whose team owns its only consumer and its only test. No file loses an owner. The legacy-files.txt entry likewise follows the implementation to the new path, so the module stays in yapf's scope and the entry count is unchanged at 765.

4) tensorrt_llm/serialization.py becomes a shim

The old path keeps a definition-free shim over all seven public names, warning on import. serialization has no leading underscore, so it is a public module path and keeps a shim regardless of what the evidence says; what the measurement earns is a shorter deprecation window, and it is what fixes the export set -- with no evidence source naming a symbol, the supported surface falls back to the module's own public surface: BASE_EXAMPLE_CLASSES, Unpickler, dump, dumps, load, loads and register_approved_class.

A shim, and only a shim, preserves `from tensorrt_llm import serialization`. That form works today only because serialization is a submodule of the package; __init__.py never imports it, so the top-level re-export that rescues other modules' symbols cannot help here -- what is imported is the module itself, and that goes through the import finder.

Because the shim re-exports objects rather than copies, isinstance and pickle keep working: every name satisfies
`tensorrt_llm.serialization.X is tensorrt_llm.llmapi.serialization.X`, seven of seven, nothing is defined at the old path and no __module__ is rewritten. The two module objects are necessarily distinct -- a re-export shim shares symbols, not identity -- so per-symbol identity is the assertion that holds, and it is the property pickle and isinstance actually depend on. The security boundary forwards too, not just the names: a payload dumped through either path loads through the other, and an unapproved class is still rejected when loaded through the shim. A shim that forwarded only the success path would quietly widen the boundary while every round-trip test passed.

5) .pre-commit-config.yaml, pyproject.toml, ruff-legacy.toml

Regenerated from legacy-files.txt, so no derived config still names the retired path; check-configs passes and the ruff baseline needs no key renames. Both shims are new files and therefore not on the legacy list, which puts them in ruff's scope; their formatting is already what ruff produces, so the hooks leave them alone.

Left alone deliberately

version.py stays at the package root: setup.py, lock-file generation and the packaging metadata read that exact source path. _deprecation.py is confirmed absent at HEAD with no remaining reference.

Nothing outside the two deprecation warnings changes for callers: the top-level __all__ is byte-identical and hashes identically at runtime, the api-stability reference diff is empty, the golden manifest's sha256 is unchanged, no test moves so node IDs, test lists and waives.txt are untouched, and the import graph keeps its five module-level cycles with no new cross-package edge and neither shim joining one. Against a build of the base commit with this change overlaid, test_serialization.py passes 6 before and after and pre-migration pickles replay 10 of 10. The eleventh class in that compatibility matrix, GreedyDecodingParams, does not exist at this commit -- the name appears nowhere in the repository except as a string in the allowlist. Left as-is and recorded for the task that updates it.

## Dev Engineer Review

- Consolidates package bootstrap logic in `tensorrt_llm/_bootstrap.py`.
- Preserves bootstrap phase ordering and defers phase-two imports inside `_init()`.
- Keeps `tensorrt_llm/__init__.py` as the package compatibility surface.
- Converts `tensorrt_llm/_common.py` into a deprecation shim.
- Moves restricted serialization to `tensorrt_llm/llmapi/serialization.py`.
- Preserves the top-level serialization API through a deprecation shim.
- Updates consumers, comments, ownership rules, formatting configuration, and legacy-file configuration.
- `version.py` remains unchanged.
- Reported checks show unchanged exports, API-stability references, manifest hash, test identifiers, and import-cycle count.
- Pre-migration pickles replay successfully for the tested compatibility matrix.
- Both deprecated paths emit warnings on direct import.

## QA Engineer Review

- Modified test file: `tests/unittest/llmapi/test_serialization.py`.
- Test functions present:
  - `test_serialization_allowed_class`
  - `test_serialization_disallowed_class`
  - `test_serialization_basic_object`
  - `test_serialization_complex_object_allowed_class`
  - `test_serialization_complex_object_partially_allowed_class`
  - `test_serialization_complex_object_disallowed_class`
- No test-list, `test-db/`, `qa/`, or `waives.txt` changes were reported.
- Coverage for these test functions in `tests/integration/test_lists/` is unavailable.
- Verdict: needs follow-up.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: legacy-files.txt [hunk 1]
```diff
@@ -394,6 +394,7 @@ tensorrt_llm/llmapi/mgmn_worker_node.py
 tensorrt_llm/llmapi/mm_encoder.py
 tensorrt_llm/llmapi/mpi_session.py
 tensorrt_llm/llmapi/reasoning_parser.py
+tensorrt_llm/llmapi/serialization.py
```
Comment (QiJune) [inline legacy-files.txt:400]: Since this implementation is moving to a new canonical path, could we graduate it from legacy-files.txt instead of carrying the legacy status forward?

Comment (YihuiLu512) [reply legacy-files.txt:400]: Fair point — the entry is carried forward deliberately, not by inertia.

Graduating isn't a line deletion: it's the five-step procedure in CODING_GUIDELINES.md, and step 4 is a real reformat. Measured with the pinned ruff 0.9.4 and this repo's config, 211 of the file's 224 lines change. Today the file at the new path differs from the pre-move original by exactly the 14-line copyright header — the property that makes relocating a restricted unpickler reviewable at a glance. I'd rather not trade that for a 211-line diff in the same commit, on a module whose content is a security allowlist.

Proposal: this phase moves code and verifies references only. Graduating legacy-files.txt entries is better done as a batch in a later phase, in one epic together with code simplification and dead-code removal — the same files get touched, the reformat noise lands once, and each PR keeps a single concern.
